hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java

```diff
@@ -671,7 +727,7 @@ public Builder setNetworkName(String name) {
      * @return DatanodeDetails.Builder
      */
     public Builder setNetworkLocation(String loc) {
-      this.networkLocation = loc;
+      this.networkLocation = new StringWithByteString(NetUtils.normalize(loc));
```

Comment:
Since the `loc` is likely to be already normalized, we may add another  `setNetworkLocation(String loc, ByteString locBytes)` method as below:
```java
    public Builder setNetworkLocation(String loc) {
      return setNetworkLocation(loc, null);
    }

    public Builder setNetworkLocation(String loc, ByteString locBytes) {
      final String normalized = NetUtils.normalize(loc);
      this.networkLocation = normalized.equals(loc) && locBytes != null
          ? new StringWithByteString(normalized, locBytes)
          : new StringWithByteString(normalized);
      return this;
    }
```